### PR TITLE
Fix #473

### LINKED ISSUE #473: "sh line 2: warning: here-document at line 0 delimited by end-of-file (wanted `MWexnsqihH')" displayed when deleting characters continuously

### PR DESCRIPTION
**Pull request by fox0430**
Close #473

### CODE REVIEW

**Comment by walkre-niboshi** [comment]
Following steps could still cause this issue.

1. Start moe.
1. Enter 'i' to switch to insert mode.
1. Enter '`' (backquote).
1. Press the ESC key to switch to normal mode.
1. Enter 'x'

**Comment by fox0430** [comment]
@walkre-niboshi 
Thank you.
I will fix it.

File: src/moepkg/normalmode.nim [hunk 1]
```diff
@@ -430,22 +430,24 @@ proc genDelimiterStr(buffer: string): string =
     if buffer != result: break
 
 proc sendToClipboad*(registers: Registers, platform: Platform) =
-  var buffer = if registers.yankedStr.len > 0: $registers.yankedStr else: $registers.yankedLines
+  let buffer = if registers.yankedStr.len > 0: $registers.yankedStr else: $registers.yankedLines
   if buffer.len < 1: return
 
-  if buffer in "`":
+  var sendStr = ""
```
Comment (walkre-niboshi) [inline src/moepkg/normalmode.nim]: Maybe you can make the code shorter as below.
```
let sendStr = buffer.replace("`", "\\`")
```

**Comment by walkre-niboshi** [comment]
Let me present another example which involves with viaual mode.

1. Start moe, redirecting stderr to a file (for example `./moe 2> error.txt`).
1. Enter 'i' to switch to insert mode.
1. Enter '\' and '`'.
1. Press the ESC key to switch to normal mode.
1. Move to the first character('\').
1. Enter 'v' to switch to visual mode.
1. Move to the last character.
1. Enter 'x' to delete the all characters.
1. Exit moe, and see the file you redirected stderr.

In my environment, by following the above steps I can see the error: `sh: 4: Syntax error: EOF in backquote substitution`.

By the way

**Comment by walkre-niboshi** [comment]
I suggest that we quote the the delimiting identifier of the here document.

**Comment by walkre-niboshi** [comment]
In the first place, we shouldn't use a procedure which just executes a command such as `osproc.execProcess proc` instead of `os.execShellCmd` ?

**Comment by fox0430** [comment]
## TODO
Add unit test

**Pull request merged**